### PR TITLE
Update load_balancing_weight comments

### PR DESCRIPTION
**Pull request by hzxuzhonghu**
Ref: https://github.com/envoyproxy/envoy/issues/1285#issuecomment-409389357

### CODE REVIEW

File: api/envoy/api/v2/endpoint/endpoint.proto [hunk 1]
```diff
@@ -106,12 +106,6 @@ message LocalityLbEndpoints {
   // configured. These weights are ignored otherwise. If no weights are
   // specificed when locality weighted load balancing is enabled, the cluster is
   // assumed to have a weight of 1.
-  //
-  // .. attention::
-  //
-  //   The limit of 128 is somewhat arbitrary, but is applied due to performance
-  //   concerns with the current implementation and can be removed when
-  //   `this issue <https://github.com/envoyproxy/envoy/issues/1285>`_ is fixed.
```
Comment (mattklein123) [inline api/envoy/api/v2/endpoint/endpoint.proto]: Should we remove the 128 max below at this point? If we do this I think it's also worthy of a release note? @htuch?

Comment (htuch) [reply api/envoy/api/v2/endpoint/endpoint.proto]: This restriction needs to live on given how we're doing weighted Maglev; there's a tradeoff between table size and precision.

Comment (mattklein123) [reply api/envoy/api/v2/endpoint/endpoint.proto]: I would need to page all the Maglev stuff back in. So should we update the comment here to talk about the max in relation to Maglev?

Comment (htuch) [reply api/envoy/api/v2/endpoint/endpoint.proto]: Yeah, for EDF scheduler and the like, I don't think this needs a limit. For anything with a fixed table size, it becomes an issue. I worry about making this unbounded though, since we might want to move from EDF to a fixed schedule for WRR in the future, and making this unbounded would make the size of that data structure unbounded. Do we need to raise the limit?

FYI for Maglev, see https://github.com/envoyproxy/envoy/pull/3004. We compress down two tables into one, which makes this quite a bit worse than it needs to be.

Comment (hzxuzhonghu) [reply api/envoy/api/v2/endpoint/endpoint.proto]: I am not familiar with the weighted Maglev and table size.  What is the limit for Maglev ? And any limit for others?

Comment (htuch) [reply api/envoy/api/v2/endpoint/endpoint.proto]: Just keep it at 128 for now I think for all.

Comment (hzxuzhonghu) [reply api/envoy/api/v2/endpoint/endpoint.proto]: What if a value exceed 128 passed in?

Comment (htuch) [reply api/envoy/api/v2/endpoint/endpoint.proto]: The existing proto constraint ensures that the config is rejected in that case.

Comment (hzxuzhonghu) [reply api/envoy/api/v2/endpoint/endpoint.proto]: Thanks for clarification. But I have a doubt: if we have more endpoints A (>128) in a locality, how can I configure the weight? Also I have another locality of B(128) endpoints. 

If I set both of them 128, then that's not well weighted. Should I calculate a valid value, like A/B=3, then I can set 3 and 1?

Comment (htuch) [reply api/envoy/api/v2/endpoint/endpoint.proto]: I'm not sure of the exact use case here; do you need to have every endpoint have a unique weight? This is pretty unusual for weighted LB; usually you have equivalence classes of endpoints and they share weights.

Comment (hzxuzhonghu) [reply api/envoy/api/v2/endpoint/endpoint.proto]: yes, there is a not equivalence number endpoints in each locality.

Comment (hzxuzhonghu) [reply api/envoy/api/v2/endpoint/endpoint.proto]: In `ClusterLoadAssignment.Endpoints`, how can I set each `LocalityLbEndpoints.LoadBalancingWeight`
https://github.com/envoyproxy/go-control-plane/blob/master/envoy/api/v2/eds.pb.go#L49

Comment (htuch) [reply api/envoy/api/v2/endpoint/endpoint.proto]: @hzxuzhonghu not sure about what is being asked here, can you rephrase?

Comment (hzxuzhonghu) [reply api/envoy/api/v2/endpoint/endpoint.proto]: I mean should calculate the relative weight in more than one Endpoints scenario.
